server/src/main/kotlin/org/javacs/kt/KotlinWorkspaceService.kt

```diff
@@ -53,6 +58,26 @@ class KotlinWorkspaceService(
                     )
                 )))))
             }
+
+            RESOLVE_MAIN -> {
```

Comment:
Is there an advantage to making resolution of the main class a command rather than e.g. a non-standard request (something like `kotlin/mainClass`)? While the LSP spec permits returning arbitrary data in command requests, my understanding is that commands are intended for actions that perform workspace edits (see [here](https://microsoft.github.io/language-server-protocol/specification#workspace_executeCommand)). Using a non-standard request would be as easy as adding a method, since we have most of the infrastructure for that already (see `KotlinProtocolExtensions`).

Reply:
Don't really have a strong opinion here, I just thought command made sense. When reading your comment, a non-standard request makes more sense 🙂  (guess I forgot about the mention on workspace edits in the specs). Have changed according to your comment in my latest commit.